### DISCUSSION TITLE
A smarter way for transform "angle-bracket syntax" type cast in .ts and .tsc file when tsx is `true`

### DISCUSSION THREAD

**Discussion by IronLu233** [swc-project/swc#1916] [Ideas]
``` ts
// type-cast.ts
(<any>window).logger = mainLogger;
```
this code can not be compiled when `jsc.parser.tsx` is `true` now.

But in `tsc` compiler, although `compilerOptions.react` is true, the compile execution works well because the file extension is `ts`, not `tsx`.

So we can improve the parser, when file extension is `ts`, support `angle-bracket syntax` although `jsc.parser.tsx` is `true`

**Comment by kdy1** [comment]
You can use multiple entry to configure it, but I think detecting `.ts` extension would worth. (As it's dead simple)
